spec/aliases.rst

```diff
@@ -0,0 +1,189 @@
+Type aliases
+============
+
+(See :pep:`613` for the introduction of ``TypeAlias``, and
+:pep:`695` for the ``type`` statement.)
+
+Type aliases may be defined by simple variable assignments::
+
+  Url = str
+
+  def retry(url: Url, retry_count: int) -> None: ...
+
+Or by using ``typing.TypeAlias``::
+
+  from typing import TypeAlias
+
+  Url: TypeAlias = str
+
+  def retry(url: Url, retry_count: int) -> None: ...
+
+Or by using the ``type`` statement (Python 3.12 and higher)::
+
+  type Url = str
+
+  def retry(url: Url, retry_count: int) -> None: ...
+
+Note that we recommend capitalizing alias names, since they represent
+user-defined types, which (like user-defined classes) are typically
+spelled that way.
+
+Type aliases may be as complex as type hints in annotations --
+anything that is acceptable as a type hint is acceptable in a type
+alias::
+
+    from typing import TypeVar
+    from collections.abc import Iterable
+
+    T = TypeVar('T', int, float, complex)
+    Vector = Iterable[tuple[T, T]]
+
+    def inproduct(v: Vector[T]) -> T:
+        return sum(x*y for x, y in v)
+    def dilate(v: Vector[T], scale: T) -> Vector[T]:
+        return ((x * scale, y * scale) for x, y in v)
+    vec: Vector[float] = []
+
+
+This is equivalent to::
+
+    from typing import TypeVar
+    from collections.abc import Iterable
+
+    T = TypeVar('T', int, float, complex)
```

Comment:
```suggestion
    T = TypeVar('T')
```

Reply:
I think that's too simple a change because these functions wouldn't typecheck with an unconstrained TypeVar.

Reply:
That's a fair point, how about just `bound=float` for now? That's simple enough and should still work.

Reply:
Yes, that's good.